src/common/gui/MainCreateButton.ts

```diff
@@ -0,0 +1,31 @@
+import { lang, TranslationKey } from "../misc/LanguageViewModel.js"
+import { ClickHandler } from "./base/GuiUtils.js"
+import m, { Children } from "mithril"
+import { theme } from "./theme.js"
+import { px, size } from "./size.js"
+import { BaseButton, BaseButtonAttrs } from "./base/buttons/BaseButton.js"
+
+export interface MainCreateButtonAttrs {
+	label: TranslationKey
+	click: ClickHandler
+	class?: string
+}
+
+/**
+ * Main button used to open the creation dialog for emails,contacts and events.
+ */
+export const MainCreateButton = (buttonAttrs: MainCreateButtonAttrs): Children => {
```

Comment:
this will not create a mithril component with a vnode that's associated with `MainCreateButton` which might lead to surprising behavior.

If you want to avoid boilerplate consider using `pureComponent()` but since you anyway have a named type for your attrs I don't see much benefit in doing that over just writing a class.